server/mqtt.go

```diff
@@ -2164,6 +2164,30 @@ func (as *mqttAccountSessionManager) removeSession(sess *mqttSession, lock bool)
 	}
 }
 
+// Helpers that sets the sub's mqtt fields and possibly serialize
+// (pre-loaded) retained messages.
+// Account manager and session lock held.
```

Comment:
Not true for "account manager" in all cases (when called in processJSConsumer, only session). So could be changed to "Session lock held. Account manager lock possibly held".

Reply:
Account lock does not matter to this function; changed the comment.